tests/catch/CMakeLists.txt

```diff
@@ -231,7 +231,7 @@ function(catch_executable src)
 
     message("Standalone Catch test: ${exec_name} ${src}")
     add_executable(${exec_name} EXCLUDE_FROM_ALL  ${src} $<TARGET_OBJECTS:Main_Object_Standalone> $<TARGET_OBJECTS:KERNELS>) 
-    target_link_libraries(${exec_name} PRIVATE stdc++fs -pthread )
+    target_link_libraries(${exec_name} PRIVATE -pthread )
```

Comment:
`stdc++fs` is spammed all over the HIP codebases, despite often requiring c++17 or newer and with bad conditions.

These changes might break it for some users, but rather than adding it manually, we should be using CMake to add filesystem support, OR, require c++17 through CMake and then fail.